File: tools/dbench/answers/python_sonnet/s2/m1_oldest.py
best = None
with open("users.txt") as f:
    for line in f:
        p = line.split()
        if len(p) == 3:
            try:
                age = int(p[2])
                if best is None or age > best[1]:
                    best = (p[1], age)
            except ValueError:
                pass
print(f"{best[0]} {best[1]}")

File: tools/dbench/answers/rust_sonnet/s2/m1_oldest.rs
use std::fs;

fn main() {
    let contents = fs::read_to_string("users.txt").unwrap_or_default();
    let best = contents.lines()
        .filter_map(|line| {
            let parts: Vec<&str> = line.split_whitespace().collect();
            if parts.len() == 3 {
                parts[2].parse::<i64>().ok().map(|age| (parts[1].to_string(), age))
            } else {
                None
            }
        })
        .max_by_key(|(_, age)| *age);
    if let Some((name, age)) = best {
        println!("{} {}", name, age);
    }
}
